### ISSUE TITLE
[BEIT-3] How to do intermediate finetune (ITC)?

### ISSUE THREAD

**Issue by sjtulyf123** [microsoft/unilm#1347]
Dear authors, I wonder how can we conduct the intermediate finetune (actually ITC task) as you shown in the appendix.

1. Is the process is (step i) do MVLM and get pretrained model like BEiT3-base; (step ii) do ITC on the pretrained model in (i) and get model like BEiT3-base-itc ?

2. How to train the ITC part? I wonder which of the three methods below is correct, or the official implementation use another method?

Method 1: Concat image and text and input them to beit3 together, and then use CLS (position 0) and BOS (position multiway_split_position) to do ITC? (use fusion encoder)

Method 2: Input image only and get its CLS, input text only and get its CLS. Then use these two CLS to do ITC (use dual encoder)?

Method 3: Concat image and empty text and input them to beit-3, then get CLS. Concat text and empty image and input them to beit-3, then get CLS. Finally use these two CLS to do ITC? (also use fusion encoder)

3. Based on question 2, in the inference (feature generation) step, how to generate image and text features? concat together or separately input?

**Comment by wenhui0924** [comment]
Hi @sjtulyf123,

1. Yes, we perform ITC on the pretrained BEiT-3 model, which is trained using masked data modeing.
2. We use Method 2.
3. For inference, it depends on the tasks. For retrieval tasks, we use BEiT-3 as a dual encoder. For tasks require deep fusion, we use BEiT-3 as a fusion encoder.

**Comment by sjtulyf123** [comment]
> Hi @sjtulyf123,
> 
> 1. Yes, we perform ITC on the pretrained BEiT-3 model, which is trained using masked data modeing.
> 2. We use Method 2.
> 3. For inference, it depends on the tasks. For retrieval tasks, we use BEiT-3 as a dual encoder. For tasks require deep fusion, we use BEiT-3 as a fusion encoder.

Thanks for your reply.

I also wonder in question 1, when performing ITC, do you still keep training the masked data modeling task? Or only performing the ITC task?

In question 3, after getting the pretrained ITC model, do you use the projected results after CLS as the final image or text feature, or just use the CLS as the feature?

**Comment by wenhui0924** [comment]
1. We only perform the ITC task.
2.  For retrieval tasks, we use projected results after CLS.
